android/src/main/java/com/media4care/twilio/video/TwilioVideoPlugin.java

```diff
@@ -24,20 +24,30 @@ public void load() {
 
     @PluginMethod
     public void joinTwilioRoom(PluginCall call) {
-        Intent intent = new Intent(getContext(), TwilioVideoActivity.class);
-        intent.putExtra("roomName", call.getString("roomName"));
-        intent.putExtra("accessToken", call.getString("accessToken"));
-        intent.putExtra("options", call.getObject("options").toString());
+        try {
+            Intent intent = new Intent(getContext(), TwilioVideoActivity.class);
+            intent.putExtra("roomName", call.getString("roomName"));
+            intent.putExtra("accessToken", call.getString("accessToken"));
+            intent.putExtra("options", call.getObject("options").toString());
 
-        getContext().startActivity(intent);
-        call.success();
+            getContext().startActivity(intent);
+            call.success();
+        } catch (Exception e) {
+            e.printStackTrace();
+            call.reject(e.getMessage());
+        }
     }
 
     @PluginMethod
     public void leaveTwilioRoom(PluginCall call) {
-        Intent intent = new Intent(TwilioVideoActivity.CLOSE_EVENT);
-        LocalBroadcastManager.getInstance(getContext()).sendBroadcast(intent);
-        call.success();
+        try {
+            Intent intent = new Intent(TwilioVideoActivity.CLOSE_EVENT);
+            LocalBroadcastManager.getInstance(getContext()).sendBroadcast(intent);
+            call.success();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Same as previous comment any exception in the intent will not be caught here

Reply:
removed